File: backend/src/main/scala/admin/api/RejectMerchantApplicationApi.scala
package admin.api

import domain.shared.given

import cats.effect.IO
import domain.merchant.ReviewMerchantApplicationRequest
import domain.shared.{DeliveryAppState, MerchantApplicationId, UserRole}
import merchant.app.*
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import shared.api.routing.*

val rejectMerchantApplicationApi: FixedMethodApi1[MerchantApplicationId, DeliveryAppState] =
  jsonPostApi1[MerchantApplicationId, ReviewMerchantApplicationRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("merchant-applications")),
    List(routeSegment("reject")),
  )

File: frontend/src/apis/admin/RejectMerchantApplicationApi.ts
import type { DeliveryAppState, MerchantApplicationId, ReviewMerchantApplicationRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const rejectMerchantApplicationApiDefinition = defineJsonPostApi1<MerchantApplicationId, ReviewMerchantApplicationRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('merchant-applications')],
    [routeSegment('reject')],
  )

export function rejectMerchantApplication(
  applicationId: MerchantApplicationId,
  payload: ReviewMerchantApplicationRequest,
) {
  return postNormalizedDeliveryState(buildApiPath1(rejectMerchantApplicationApiDefinition, applicationId), payload)
}
